packages/core/src/command/hl7v2-subscriptions/hl7v2-roster-generator.ts

```diff
@@ -0,0 +1,220 @@
+import { MetriportError, errorToString, executeWithNetworkRetries } from "@metriport/shared";
+import axios from "axios";
+import { stringify } from "csv-stringify/sync";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import _ from "lodash";
+import Client from "ssh2-sftp-client";
+import { Hl7v2Subscriber, Hl7v2Subscription } from "../../domain/patient-settings";
+import { S3Utils, storeInS3WithRetries } from "../../external/aws/s3";
+import { capture, out } from "../../util";
+import { Config } from "../../util/config";
+import { CSV_FILE_EXTENSION, CSV_MIME_TYPE } from "../../util/mime";
+import {
+  Hl7v2RosterConfig,
+  Hl7v2SubscriberApiResponse,
+  Hl7v2SubscriberParams,
+  SftpConfig,
+} from "./get-subscribers";
+
+dayjs.extend(duration);
+
+const region = Config.getAWSRegion();
+
+type SubscriberRecord = Record<string, string>;
+
+type RosterGenerateProps = {
+  config: Hl7v2RosterConfig;
+  bucketName: string;
+  apiUrl: string;
+};
+
+const HL7V2_SUBSCRIBERS_ENDPOINT = `internal/patient/hl7v2-subscribers`;
+const NUMBER_OF_PATIENTS_PER_PAGE = 500;
+const NUMBER_OF_ATTEMPTS = 3;
+const BASE_DELAY = dayjs.duration({ seconds: 1 });
+
+function getS3UtilsInstance(): S3Utils {
+  return new S3Utils(region);
+}
+
+export function convertSubscribersToHieFormat(
+  subscribers: Hl7v2Subscriber[],
+  schema: Record<string, string>
+): SubscriberRecord[] {
+  const convertToOutgoingSchema = (s: Hl7v2Subscriber) => convertSubscriberToHieFormat(s, schema);
+  return subscribers.map(convertToOutgoingSchema);
+}
+
+export function convertSubscriberToHieFormat(
+  subscriber: Hl7v2Subscriber,
+  schema: Record<string, string>
+): SubscriberRecord {
+  const result: SubscriberRecord = {};
+
+  for (const [ourField, hieField] of Object.entries(schema)) {
+    const value = _.get(subscriber, ourField);
+    if (value !== undefined) {
+      result[hieField] = String(value);
+    }
+  }
+
+  return result;
+}
+
+export async function generateAndUploadHl7v2Roster({
```

Comment:
Can we split this into smaller functions?
- build roster
- store on S3
- send

Find to do as a follow-up, though